include/zephyr/sys_clock.h

```diff
@@ -289,6 +289,25 @@ static inline bool sys_timepoint_expired(k_timepoint_t timepoint)
 	return K_TIMEOUT_EQ(sys_timepoint_timeout(timepoint), Z_TIMEOUT_NO_WAIT);
 }
 
+/**
+ * @brief Compare two timepoint values.
+ *
+ * This function is used to compare two timepoint values and find the one that is pointing
+ * to earlier point in time. It does not verify if given timepoints are already expired.
+ *
+ * @param a Timepoint to compare
+ * @param b Timepoint to compare against.
+ * @return zero if both timepoints are the same. Negative value if timepoint A points to earlier
+ * time, positive otherwise.
+ */
+static inline int sys_timepoint_cmp(k_timepoint_t a, k_timepoint_t b)
+{
+	if (a.tick == b.tick) {
+		return 0;
+	}
+	return a.tick < b.tick ? -1 : 1;
+}
```

Comment:
This won't compile if `CONFIG_SYS_CLOCK_EXISTS==n` as the `tick` member would be replaced with `wait`.

Reply:
How do I keep missing that.... On the rebase this `#if .. # else`  was added.
So I added this into !CLOCK_EXISTS:
```c
static inline int sys_timepoint_cmp(k_timepoint_t a, k_timepoint_t b)
{
	if (a.wait == b.wait) {
		return 0;
	}
	return b.wait ? -1 : 1;
}
```
I believe it should be right as the `wait` is boolean.